pkg/storegateway/bucket.go

```diff
@@ -777,6 +783,61 @@ func storeCachedSeries(ctx context.Context, indexCache indexcache.IndexCache, us
 	indexCache.StoreSeries(ctx, userID, blockID, entry.MatchersKey, shard, data)
 }
 
+type seriesForPostingsCacheEntry struct {
+	LabelSets   []labels.Labels
+	MatchersKey indexcache.LabelMatchersKey
+	Shard       sharding.ShardSelector
+}
+
+func fetchCachedSeriesForPostings(ctx context.Context, userID string, indexCache indexcache.IndexCache, blockID ulid.ULID, matchers []*labels.Matcher, shard *sharding.ShardSelector, postingsKey indexcache.PostingsKey, logger log.Logger) (seriesChunkRefsSet, bool) {
+	matchersKey := indexcache.CanonicalLabelMatchersKey(matchers)
+	data, ok := indexCache.FetchSeriesForPostings(ctx, userID, blockID, matchersKey, shard, postingsKey)
+	if !ok {
+		return seriesChunkRefsSet{}, false
+	}
+	var entry seriesForPostingsCacheEntry
+	if err := decodeSnappyGob(data, &entry); err != nil {
+		level.Warn(spanlogger.FromContext(ctx, logger)).Log("msg", "can't decode series cache", "err", err)
```

Comment:
This and the following logs don't give us information to investigate it, because it miss information to construct the whole cache key. I would log all info here and in the following logs.